api/src/main/java/bio/terra/tanagra/service/search/SqlVisitor.java

```diff
@@ -294,19 +296,89 @@ public String resolveTable(EntityVariable entityVariable) {
       return String.format("%s AS %s", resolvedTable, entityVariable.variable().name());
     }
 
+    /** Resolve a {@link Table} into a SQL string. */
     private String resolveTable(Table table) {
       // `projectId.datasetId`.table
       return String.format(
           "`%s.%s`.%s", table.dataset().projectId(), table.dataset().datasetId(), table.name());
     }
 
+    /** Resolve a {@link Table} with a {@link TableFilter} into a SQL string. */
     private String resolveTable(Table table, @Nullable TableFilter tableFilter) {
       if (tableFilter == null) {
         return resolveTable(table);
+      } else if (tableFilter.arrayColumnFilter() != null) {
+        return resolveArrayColumnFilter(table, tableFilter.arrayColumnFilter(), false);
+      } else if (tableFilter.binaryColumnFilter() != null) {
+        return resolveBinaryColumnFilter(table, tableFilter.binaryColumnFilter(), false);
+      } else {
+        throw new IllegalArgumentException(
+            "Invalid table filter missing binary and array column filters.");
+      }
+    }
+
+    /**
+     * Resolve a {@link Table} with an {@link ArrayColumnFilter} into a SQL string.
+     *
+     * @param whereClauseOnly If true, only returns the part of the SQL string after the WHERE. This
+     *     is helpful when building an {@link ArrayColumnFilter} composed of multiple sub-filters.
+     */
+    private String resolveArrayColumnFilter(
+        Table table, ArrayColumnFilter arrayColumnFilter, boolean whereClauseOnly) {
+      String joinOperatorInWhereClause;
+      switch (arrayColumnFilter.operator()) {
+        case AND:
+          joinOperatorInWhereClause = "AND";
+          break;
+        case OR:
+          joinOperatorInWhereClause = "OR";
+          break;
+        default:
+          throw new IllegalArgumentException(
+              "Unknown array column filter operator type: " + arrayColumnFilter.operator());
+      }
+
+      // %s AND %s AND (%s OR %s) ...
+      StringBuilder joinedWhereClauses = new StringBuilder();
+      int subFiltersCtr = 0;
+      for (BinaryColumnFilter subFilter : arrayColumnFilter.binaryColumnFilters()) {
```

Comment:
opt nit: streams + joiner could avoid the need for the `ctr` variable

```
Streams.concat(
  arrayColumnFilter.binaryColumnFilters().stream().map(f -> resolveBinaryColumnFilter(table, f, true)),
  arrayColumnFilter.arrayColumnFilters().stream().map(f -> resolveArrayColumnFilters(table, f, true))
).collect(Collectors.joining(" " + joinOperatorInWhereClause + " "))
```

Reply:
Good tip, thanks! Modified slightly to avoid appending the join operator at the beginning.

Reply:
Hmm perhaps I'm missing something.. a joiner only places the separator between elements so the first element shouldn't need any special handling.

Reply:
Oh you're totally right. I just realized that when I tried this I wasn't updating the original `StringBuilder` variable, so I need to just skip that the return the result of this `collect()`. Thanks for the correction! I'll add this change to one of the existing open PRs.